lightningd/param.c

```diff
@@ -79,7 +79,7 @@ static bool parse_by_position(struct command *cmd,
 	}
 
 	/* check for unexpected trailing params */
-	if (tok != end) {
+	if (!cmd->allow_unused && tok != end) {
 		command_fail(cmd, JSONRPC2_INVALID_PARAMS,
 			     "too many parameters:"
 			     " got %u, expected %zu",
```

Comment:
Can we modify parse_by_name similarly?

Currently:
```
$ ./cli/lightning-cli check command_to_check=listpeers id=xxx
{ "code" : -32602, "message" : "unknown parameter: 'id'" }
```

Reply:
oops.